lib/internal/test_runner/reporter/spec.js

```diff
@@ -35,114 +35,116 @@ const symbols = {
   'arrow:right': '\u25B6 ',
   'hyphen:minus': '\uFE63 ',
 };
-class SpecReporter extends Transform {
-  #stack = [];
-  #reported = [];
-  #indentMemo = new SafeMap();
-  #failedTests = [];
+const stack = [];
+const reported = [];
+const indentMemo = new SafeMap();
+const failedTests = [];
```

Comment:
Doesn't this mean that calling `run` twice with `spec` now shares the failures between runs? (whereas previously you could instantiate a new reporter as per need)

Reply:
I think I would expect `const spec` to be a function creating these vars and return the `new Transform`. Does that make sense?

Reply:
makes sense 👍🏻 will change

Reply:
Hey @atlowChemi, I have updated the PR with the lates changes. Instead of using the `new Transform`, I have modified it to an async generator, similar to `dot` and `tap` reporters.